pkg/ddl/ingest/disk_root.go

```diff
@@ -155,3 +193,20 @@ func (d *diskRootImpl) StartupCheck() error {
 func RiskOfDiskFull(available, capacity uint64) bool {
 	return float64(available) < (1-capacityThreshold)*float64(capacity)
 }
+
+// CheckIngestLeakageForTest is only used in test.
+func CheckIngestLeakageForTest(exitCode int) {
+	if exitCode == 0 {
+		leakObj := ""
+		if TrackerCountForTest.Load() != 0 {
+			leakObj = "disk usage tracker"
+		} else if BackendCounterForTest.Load() != 0 {
+			leakObj = "backend context"
+		}
+		if len(leakObj) > 0 {
+			fmt.Fprintf(os.Stderr, "add index leakage check failed: %s leak\n", leakObj)
+			os.Exit(1)
+		}
+	}
+	os.Exit(exitCode)
```

Comment:
```suggestion
```
seems not needed, let TestMain handle the exit

Reply:
```

// Variables for stubbing in unit tests.
var (
	_osExit             = os.Exit
	_osStderr io.Writer = os.Stderr
)
```
```
	var cleanup func(int)
	cleanup, opts.cleanup = opts.cleanup, nil
	if cleanup == nil {
		cleanup = _osExit
	}
	defer func() { cleanup(exitCode) }()
```

Cleanup function should be a replacement of `os.Exit`

Reply:
yes, but if exitCode==0 as in this case, it's not necessary